flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java

```diff
@@ -220,16 +220,25 @@ public Asset getAssetByName(@PathVariable String name) {
       long total = assetRepository.getNumber(keyWords+"%", AssetCategory.Server.name());
       PageImpl<Asset> assetPage = new PageImpl<Asset>(assets,pageable,total);
       HashMap<String, String> assetSourceIDAndAssetSourceNameMap = new HashMap<String, String>();
+
       for (Asset asset : assetPage.getContent()) {
          String assetSourceID = asset.getAssetSource();
-         if (assetSourceID != null) {
-            String assetSourceName = assetSourceIDAndAssetSourceNameMap.get(assetSourceID);
+         if(assetSourceID == null) {
```

Comment:
For each asset source you only need to query once instead of query multiple times.

Reply:
For example, you have 100 assets in the page, all the assets come from the same source, then you just need to query it one time instead of 100 times.

Reply:
I used a map to store the mapping of Id and name, if next id is same as the id which is in the map, we can get the value from the map by the id.